dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -749,6 +749,7 @@ internal static OpenAIClientOptions GetOpenAIClientOptions(HttpClient? httpClien
         {
             options.Transport = new HttpClientTransport(httpClient);
             options.RetryPolicy = new RetryPolicy(maxRetries: 0); // Disable Azure SDK retry policy if and only if a custom HttpClient is provided.
+            options.Retry.NetworkTimeout = Timeout.InfiniteTimeSpan; // Disable Azure SDK default timeout
```

Comment:
I'm a little bit late with the review but I would like to point out that there's a similar code over here - [AzureOpenAITextToImageService GetClientOptions](https://github.com/microsoft/semantic-kernel/blob/51ee30fcbd438472ec45f90f68be77a591e04527/dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs#L148) that needs to be updated as well.

Reply:
Created this https://github.com/microsoft/semantic-kernel/pull/5569